realm/src/main/java/io/realm/Realm.java

```diff
@@ -163,19 +163,18 @@ protected void finalize() throws Throwable {
     }
 
     /**
-     * Realm static constructor for the default Realm "default.realm".
-     * {@link #close()} must be called when you are done using the Realm instance.
-     * <p>
-     * It sets auto-refresh on if the current thread has a Looper, off otherwise.
-     *
+     * Realm static constructor for the default Realm file "default.realm".
      * This is equivalent to calling {@code Realm.getInstance(new RealmConfiguration(getContext()).build()) }.
-
-     * @param context an Android {@link android.content.Context}
-     * @return an instance of the Realm class
-     * @throws RealmMigrationNeededException The model classes have been changed and the Realm
-     *                                       must be migrated
-     * @throws RealmIOException              Error when accessing underlying file
-     * @throws RealmException                Other errors
+     *
+     * This constructor is only provided for convenience. It is recommend to use
+     * {@link #getInstance(RealmConfiguration)} or {@link #getDefaultInstance()} instead.
+     *
+     * @param context An Android {@link android.content.Context}
+     * @return An instance of the Realm class.
+     * @throws RealmMigrationNeededException If the model classes no longer match the underlying Realm
+     *                                       and it must be migrated.
+     * @throws RealmIOException              If an error happened when accessing the underlying Realm
+     *                                       file.
```

Comment:
Not anything formalised. But you are right, in this case it should probably list `IllegalArgumentException` as well.

In general it would be nice to be able to move to using `@NotNull`, `@Nullable` annotations, but unfortunately it is not easy to add those if we at some point want the same codebase for both Android and JVM Java